dev_tools/ci/size-labeler.sh

```diff
@@ -0,0 +1,184 @@
+    local -r keys_filter='with_entries(select([.key] | inside(["changes", "filename"])))'
+    change_info="$(jq_stdin "map(${keys_filter})" <<<"$(api_call "pulls/${pr}/files")")"
+
+    local files total_changes
+    readarray -t files < <(jq_stdin -c '.[]' <<<"${change_info}")
+    total_changes=0
+    for file in "${files[@]}"; do
+        local name changes
+        name="$(jq_stdin -r '.filename' <<<"${file}")"
+        for pattern in "${IGNORED[@]}"; do
+            if [[ "$name" =~ ${pattern} ]]; then
+                info "File $name ignored"
+                continue 2
+            fi
+        done
+        changes="$(jq_stdin -r '.changes' <<<"${file}")"
+        info "File $name +-$changes"
+        total_changes="$((total_changes + changes))"
+    done
+    echo "$total_changes"
+}
+
+function get_size_label() {
+    local -r changes="$1"
+    for label in "${LABELS[@]}"; do
+        local limit="${LIMITS["${label}"]}"
+        if [[ "${changes}" -lt "${limit}" ]]; then
+            echo "${label}"
+            return
+        fi
+    done
+}
+
+function prune_stale_labels() {
+    local -r pr="$1"
+    local -r size_label="$2"
+    local existing_labels
+    existing_labels="$(jq_stdin -r '.labels[] | .name' <<<"$(api_call "pulls/${pr}")")"
+    readarray -t existing_labels <<<"${existing_labels}"
+
+    local correctly_labeled=false
+    for label in "${existing_labels[@]}"; do
+        [[ -z "${label}" ]] && continue
+        # If the label we want is already present, we can just leave it there.
+        if [[ "${label}" == "${size_label}" ]]; then
+            info "Label '${label}' is correct, leaving it."
+            correctly_labeled=true
+            continue
+        fi
+        # If there is another size label, we need to remove it
+        if [[ -v "LIMITS[${label}]" ]]; then
+            info "Label '${label}' is stale, removing it."
+            api_call "issues/${pr}/labels/${label}" -X DELETE &>/dev/null
+            continue
+        fi
+        info "Label '${label}' is unknown, leaving it."
+    done
+    echo "${correctly_labeled}"
+}
+
+function main() {
+    local pr=$1
+    info "Labeling PR ${pr}"
+
+    local total_changes
+    total_changes="$(compute_changes "${pr}")"
+    info "Lines changed: ${total_changes}"
+
+    local size_label
+    size_label="$(get_size_label "$total_changes")"
+    info "Appropriate label is '${size_label}'"
+
+    local correctly_labeled
+    correctly_labeled="$(prune_stale_labels "${pr}" "${size_label}")"
+
+    if [[ "${correctly_labeled}" != true ]]; then
+        api_call "issues/${pr}/labels" -X POST -d "{\"labels\":[\"${size_label}\"]}" &>/dev/null
+        info "Added label '${size_label}'"
+    fi
+}
+
+[[ "$#" -eq 1 ]] || { error "Missing required argument: the PR number." >&2; exit 1; }
+[[ -v GITHUB_TOKEN ]] || { error "Variable GITHUB_TOKEN is not set." >&2; exit 1; }
+[[ -v GITHUB_REPOSITORY ]] || { error "Variable GITHUB_REPOSITORY is not set." >&2; exit 1; }
```

Comment:
redirection is already done in the `error` function, it has no effect here.
```suggestion
[[ -v GITHUB_TOKEN ]] || { error "Variable GITHUB_TOKEN is not set."; exit 1; }
[[ -v GITHUB_REPOSITORY ]] || { error "Variable GITHUB_REPOSITORY is not set."; exit 1; }
```